app/components/dashboard-calendar.js

```diff
@@ -57,232 +62,139 @@ export default Component.extend({
     return 1;
   }),
   calendarDate: momentFormat('selectedDate', 'YYYY-MM-DD'),
-  ourEvents: computed('mySchedule', 'fromTimeStamp', 'toTimeStamp', 'selectedSchool', 'selectedView', function(){
-    if(this.get('mySchedule')) {
-      return DS.PromiseArray.create({
-        promise: this.get('userEvents').getEvents(this.get('fromTimeStamp'), this.get('toTimeStamp'))
-      });
-    } else {
-      let deferred = RSVP.defer();
-      this.get('selectedSchool').then(school => {
-        deferred.resolve(this.get('schoolEvents').getEvents(school.get('id'), this.get('fromTimeStamp'), this.get('toTimeStamp')));
-      });
-      return DS.PromiseArray.create({
-        promise: deferred.promise
+
+  /**
+   * @property academicYears
+   * @type {Ember.computed}
+   * @public
+   */
+  academicYears: computed('allAcademicYears.[]', 'academicYearSelectedByUser', function(){
+    return new Promise(resolve => {
+      this.get('allAcademicYears').then(years => {
+        resolve(years.sortBy('title'));
       });
+    });
+  }),
+
+  /**
+   * @property cohorts
+   * @type {Ember.computed}
+   * @public
+   * @todo Check if this can be replaced with allCohorts(). [ST 2017/02/07]
+   */
+  cohorts: computed('allCohorts.[].displayTitle', 'selectedCohorts.[]', function(){
```

Comment:
Why does this CP depend on `displayTitle` and `selectedCohorts.[]` but not use them in the body fo the method?